backend/src/common/database.ts

```diff
@@ -102,6 +120,24 @@ export const addEntityToNullFields = (ontology: Ontology, entity: Node): Ontolog
   Predicate: ontology.Predicate,
   Object: ontology.Object || entity,
 });
+export const removeDuplicatePredicates = (ontology: any, sameOntology: any): any => {
```

Comment:
Instead of taking two identical arguments you can use another parameter in forEach. So for example `ontology.forEach((OntologyElement, index, restOfOntology) => ......`